test/relations/JavaScriptDynamicImport.js

```diff
@@ -0,0 +1,85 @@
+const expect = require('../unexpected-with-plugins');
+const AssetGraph = require('../../lib/AssetGraph');
+const pathModule = require('path');
+
+describe('JavaScriptDynamicImport', function() {
+  it('should detect a relation', function() {
+    const javaScript = new AssetGraph().addAsset({
+      type: 'JavaScript',
+      url: 'https://example.com/',
+      text: `
+        import('./bar/quux.js');
+      `
+    });
+
+    expect(javaScript.outgoingRelations, 'to satisfy', [
+      {
+        type: 'JavaScriptDynamicImport',
+        href: './bar/quux.js',
+        to: { url: 'https://example.com/bar/quux.js' }
+      }
+    ]);
+  });
+
+  it('should update the href of a relation', function() {
+    const javaScript = new AssetGraph().addAsset({
+      type: 'JavaScript',
+      url: 'https://example.com/',
+      text: `
+        import('./bar/quux.js');
+      `
+    });
+
+    javaScript.outgoingRelations[0].href = './blabla.js';
+    javaScript.markDirty();
+    expect(javaScript.text, 'to contain', "import('./blabla.js');");
+  });
+
+  it('should keep ./ in front of relative urls', async function() {
+    const assetGraph = new AssetGraph({
+      root: pathModule.resolve(
+        __dirname,
+        '../../testdata/relations/JavaScriptDynamicImport/relative/'
+      )
+    });
+    const [javaScript] = await assetGraph.loadAssets('index.js');
+    await assetGraph.populate();
+    expect(javaScript.text, 'to contain', `import('./main.js');`);
+    assetGraph.findAssets({ fileName: 'main.js' })[0].url = `${
+      assetGraph.root
+    }/static/foobar.js`;
```

Comment:
`assetGraph.root` always has a trailing slash.

```suggestion
    }static/foobar.js`;
```